### PR TITLE
databases: Add update major version

### PR DESCRIPTION
**Pull request by scotchneat**
This adds the `PUT /databases/$DATABASE_ID/upgrade` request.

For DBAAS-3199

### CODE REVIEW

File: specification/DigitalOcean-public.v2.yaml [hunk 1]
```diff
@@ -846,6 +846,10 @@ paths:
     put:
       $ref: 'resources/databases/databases_update_sql_mode.yml'
 
+  /v2/databases/{database_cluster_uuid}/upgrade:
```
Comment (scotchneat) [inline specification/DigitalOcean-public.v2.yaml:849]: I would have preferred this path be `/v2/databases/{database_cluster_uuid}/version` so we could also add a `GET` method to the path. 

I'll ask if it's too late to request this change.

**Comment by dweinshenker** [comment]
LGTM

File: specification/resources/databases/databases_upgrade_major_version.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  '500':
+    $ref: '../../shared/responses/server_error.yml'
+
+  default:
+    $ref: '../../shared/responses/unexpected_error.yml'
+
+x-codeSamples:
+  - $ref: 'examples/curl/databases_update_sql_mode.yml'
+  - $ref: 'examples/go/databases_update_sql_mode.yml'
```
Comment (andrewsomething) [inline specification/resources/databases/databases_upgrade_major_version.yml:49]: These are the examples for updating the SQL mode, not the version.

**Review by andrewsomething — CHANGES_REQUESTED** [review]

File: specification/resources/databases/databases_upgrade_major_version.yml (hunk 2 above)
Comment (scotchneat) [reply specification/resources/databases/databases_upgrade_major_version.yml:49]: Yep. Thanks.

**Review by andrewsomething — APPROVED** [review]
:+1:  lgtm!

**Pull request merged**